tests/Microsoft.Identity.Test.Unit.net45/CacheTests/CacheSerializationTests.cs

```diff
@@ -0,0 +1,609 @@
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Cache;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Json.Linq;
+using Microsoft.Identity.Test.Unit.CacheV2Tests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.CacheTests
+{
+    [TestClass]
+    public class CacheSerializationTests
+    {
+        private MsalAccessTokenCacheItem CreateAccessTokenItem()
+        {
+            return new MsalAccessTokenCacheItem
+            {
+                ClientId = MsalTestConstants.ClientId,
+                Environment = "env",
+                ExpiresOnUnixTimestamp = "12345",
+                ExtendedExpiresOnUnixTimestamp = "23456",
+                CachedAt = "34567",
+                HomeAccountId = MsalTestConstants.HomeAccountId,
+                IsExtendedLifeTimeToken = false,
+                NormalizedScopes = MsalTestConstants.ScopeStr,
+                Secret = "access_token_secret",
+                TenantId = "the_tenant_id",
+                RawClientInfo = null, // todo(cache): what should this be?
+                UserAssertionHash = "assertion hash", // todo(cache): what should this be
+            };
+        }
+
+        private MsalRefreshTokenCacheItem CreateRefreshTokenItem()
+        {
+            return new MsalRefreshTokenCacheItem
+            {
+                ClientId = MsalTestConstants.ClientId,
+                Environment = "env",
+                HomeAccountId = MsalTestConstants.HomeAccountId,
+                Secret = "access_token_secret",
+                RawClientInfo = null, // todo(cache): what should this be?
+            };
+        }
+
+        private MsalIdTokenCacheItem CreateIdTokenItem()
+        {
+            return new MsalIdTokenCacheItem
+            {
+                ClientId = MsalTestConstants.ClientId,
+                Environment = "env",
+                HomeAccountId = MsalTestConstants.HomeAccountId,
+                Secret = "access_token_secret",
+                TenantId = "the_tenant_id",
+                RawClientInfo = null, // todo(cache): what should this be?
+            };
```

Comment:
i think it's the useridentifers...we did something like this in msal v2 to generate the raw clientinfo-> ` Base64UrlHelpers.Encode("{\"uid\":\"" + uid + "\",\"utid\":\"" + utid + "\"}");`